tests/openedx_tagging/core/tagging/test_views.py

```diff
@@ -1949,3 +1953,392 @@ def test_delete_tag_in_taxonomy_without_subtags(self):
+        """
+        url = TAXONOMY_CREATE_IMPORT_URL
+        file = SimpleUploadedFile("taxonomy.invalid", b"invalid file content")
+        self.client.force_authenticate(user=self.staff)
+        response = self.client.post(
+            url,
+            {
+                "taxonomy_name": "Imported Taxonomy name",
+                "taxonomy_description": "Imported Taxonomy description",
+                "file": file,
+            },
+            format="multipart"
+        )
+        assert response.status_code == status.HTTP_400_BAD_REQUEST
+        assert response.data["file"][0] == "File type not supported: invalid"
+
+        # Check if the taxonomy was not created
+        assert not Taxonomy.objects.filter(name="Imported Taxonomy name").exists()
+
+    @ddt.data(
+        "csv",
+        "json",
+    )
+    def test_import_invalid_content(self, file_format) -> None:
+        """
+        Tests importing a taxonomy with an invalid file content.
+        """
+        url = TAXONOMY_CREATE_IMPORT_URL
+        file = SimpleUploadedFile(f"taxonomy.{file_format}", b"invalid file content")
+        self.client.force_authenticate(user=self.staff)
+        response = self.client.post(
+            url,
+            {
+                "taxonomy_name": "Imported Taxonomy name",
+                "taxonomy_description": "Imported Taxonomy description",
+                "file": file,
+            },
+            format="multipart"
+        )
+        assert response.status_code == status.HTTP_400_BAD_REQUEST
+        assert f"Invalid '.{file_format}' format:" in response.data
+
+        # Check if the taxonomy was not created
+        assert not Taxonomy.objects.filter(name="Imported Taxonomy name").exists()
+
+    def test_import_no_perm(self) -> None:
+        """
+        Tests importing a taxonomy using a user without permission.
+        """
+        url = TAXONOMY_CREATE_IMPORT_URL
+        new_tags = [
+            {"id": "tag_1", "value": "Tag 1"},
+            {"id": "tag_2", "value": "Tag 2"},
+            {"id": "tag_3", "value": "Tag 3"},
+            {"id": "tag_4", "value": "Tag 4"},
+        ]
+        file = self._get_file(new_tags, "json")
+
+        self.client.force_authenticate(user=self.user)
+        response = self.client.post(
+            url,
+            {
+                "taxonomy_name": "Imported Taxonomy name",
+                "taxonomy_description": "Imported Taxonomy description",
+                "file": file,
+            },
+            format="multipart"
+        )
+        assert response.status_code == status.HTTP_403_FORBIDDEN
+
+        # Check if the taxonomy was not created
+        assert not Taxonomy.objects.filter(name="Imported Taxonomy name").exists()
+
+
+@ddt.ddt
+class TestImportTagsView(ImportTaxonomyMixin, APITestCase):
+    """
+    Tests the taxonomy import tags action.
+    """
+    def setUp(self):
+        ImportTaxonomyMixin.setUp(self)
+
+        self.taxonomy = Taxonomy.objects.create(
```

Comment:
Nit: it's not a big deal, but I generally would prefer that we use the API methods `api.create_taxonomy(...)` and `api.add_tag_to_taxonomy(...)` instead of using the models directly. In this case for example, `add_tag_to_taxonomy` does [a lot of additional validation](https://github.com/openedx/openedx-learning/blob/7afd44d0e6fd78d2ab0dfa1124f670a7f23c846b/openedx_tagging/core/tagging/models/base.py#L529-L566) of the created objects, which is good to have, but isn't happening here if we just create the models directly.

The only major exception is in `test_models.py` where we're actually testing the models; in that case obviously we should use the models directly.

Reply:
Even in the `setUp` @bradenmacdonald?

A breaking change will prevent the tests from running and be trickier to diagnose, don't you think?

Reply:
Well that's the thing - the API is not supposed to have breaking changes in general. But we may change the internal models. So if this test isn't specifically testing the internal models, it should use the API which should be more stable. But again, don't worry about it for now. What you have is fine.